### ISSUE TITLE
"Database: Connected" removed in drush 7.x+ and causes database backups to fail

### ISSUE THREAD

**Issue by johnbburg** [forumone/web-starter#250]
The script in [drush.rake](https://github.com/forumone/web-starter/blob/1.1.x/lib/capistrano/tasks/drush.rake#L111) performs a check to see if it has a connection to the database before it attempts to generate a backup dump. This check is dependent on the following line being output in the `$ drush status` command:

```
 Database                        :  Connected 
```

This works fine with [drush 6.x](https://github.com/drush-ops/drush/blob/6.x/commands/core/core.drush.inc#L611), However, in drush 7.x, this line [seems to have been removed](https://github.com/drush-ops/drush/blob/7.x/commands/core/core.drush.inc#L555). Which means that for any site using continuous integration, which also uses drush 7 or greater, the test in drush.rake fails, and database backups are not generated prior to deployment.

**Comment by johnbburg** [comment]
Here is the commit where that 'Connected' status line was removed: https://github.com/drush-ops/drush/commit/0f4e59a9ceb1c95fdd4dfff925cc131fec9d83c1#diff-9e2b1896af01f89d4d97823474f12ae2

**Comment by azinck** [comment]
Shouldn't the build also then fail so as to avoid silently not taking backups?

**Comment by wwhurley** [comment]
The first deployment will not have a valid settings file so won't be able to connect to the database. So there is a case where we would want the deployment to proceed without a valid DB connection.

**Comment by azinck** [comment]
Ah, a tricky one.

**Comment by azinck** [comment]
Perhaps initial deployment should be kicked off in a more manual fashion and use a different script entirely.  Subsequent CI deployments should use a script that fails on critical things like this.

**Comment by wwhurley** [comment]
Right now the backup happens during the `deploy:publishing` stage which is before some of the other stages we need to actually deploy the files. We'd have to do some extensive restructuring of our Rake tasks to separate it out. It's probably the right solution, but that will likely have to wait until we change things to work with Capistrano 3.5+.

**Comment by johnbburg** [comment]
Just to note for anyone looking for a quick fix to this issue. Just remove

```
'Connected' == status['db-status'] &&
```

from this line in drush.rake: https://github.com/forumone/web-starter/blob/1.1.x/lib/capistrano/tasks/drush.rake#L111.

**Comment by johnbburg** [comment]
I wonder if this was removed from drush in error. The linked commit's comments seem irrelevant to the removal of that line. Maybe we should handle this by filing an issue over at https://github.com/drush-ops/drush/issues.

**Comment by johnbburg** [comment]
The folks over at the drush project responded quickly to re-add the connection status. https://github.com/drush-ops/drush/issues/2372. So this should be fixed in drush versions [8.1.6](https://github.com/drush-ops/drush/releases/tag/8.1.6) or greater. @mshade @jryden @wilsontech Just wanted to make sure you see this.
